File: app/src/main/java/com/yhjoo/dochef/adapter/SearchViewPagerAdapter.java
package com.yhjoo.dochef.adapter;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import java.util.ArrayList;
import java.util.List;

public class SearchViewPagerAdapter extends FragmentPagerAdapter {
    List<Fragment> fragments = new ArrayList<>();
    List<String> fragmentTitles = new ArrayList<>();

    public SearchViewPagerAdapter(FragmentManager Fm) {
        super(Fm);
    }

    public void addFragment(Fragment fragment, String title) {
        fragments.add(fragment);
        fragmentTitles.add(title);
    }

    @Override
    public Fragment getItem(int position) {
        return fragments.get(position);
    }

    @Override
    public int getCount() {
        return fragments.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return fragmentTitles.get(position);
    }
}

File: app/src/main/java/com/yhjoo/dochef/ui/adapter/SearchViewPagerAdapter.kt
package com.yhjoo.dochef.ui.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import java.util.*

class SearchViewPagerAdapter(Fm: FragmentManager?) : FragmentPagerAdapter(
    Fm!!
) {
    var fragments: MutableList<Fragment> = ArrayList()
    var fragmentTitles: MutableList<String?> = ArrayList()
    fun addFragment(fragment: Fragment?, title: String?) {
        fragments.add(fragment!!)
        fragmentTitles.add(title)
    }

    override fun getItem(position: Int): Fragment {
        return fragments[position]
    }

    override fun getCount(): Int {
        return fragments.size
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return fragmentTitles[position]
    }
}